website/pages/docs/nia/architecture.mdx

```diff
@@ -7,3 +7,45 @@ description: >-
 ---
 
 # Consul-Terraform-Sync Architecture
+
+Consul-Terraform-Sync is a service-oriented tool for managing
+network infrastructure near real-time. Consul-Terraform-Sync runs as a daemon
+and integrates the network topology maintained by your Consul cluster with your
+network infrastructure to dynamically secure and connect services.
+
+## 10,000 Foot View
+
+[![Consul-Terraform-Sync Architecture](/img/nia-highlevel-diagram.png)](/img/nia-highlevel-diagram.png)
+
+The diagram shows Consul-Terraform-Sync monitoring the Consul service catalog
+for updates and utilizing Terraform to update the state of the infrastructure.
+
+There are two principal aspects of Sync to know about corresponding to the
+lines to Consul and Terraform in the diagram above. The line to Consul
+represents the Watchers monitoring the state of Consul's service catalog (and
+possibly KV store) while the line to Terraform represents Tasks being run to
+update the infrastructure.
+
+## Watcher and Views
+
+Consul-Terraform-Sync monitors Consul for updates utilizing Consul's [Blocking
+Queries](/api-docs/features/blocking) whenever supported, falling back on
+polling when not. The watcher maintains a separate thread (known internally as
+a view) for each value monitored, running any Tasks that depend on that watched
+value whenever it's updated.
```

Comment:
This is new content! It might be helpful to further tie back to how this relates to / used by Sync, like for services, updates on IP address, health status, metadata, etc.

Reply:
How about something really simple like... "Say, for example, running a task to update a proxy when an instance goes unhealthy."

Reply:
I like it.

Reply:
Super speedy. Thanks @findkim.